server/src/instant/db/instaql.clj

```diff
@@ -1901,46 +1914,106 @@
+                                                        :type :auth
+                                                        :etype "$users"}
+                                                       current-user)
+                                 "data" (cel/->cel-map {:ctx ctx
+                                                        :type :data
+                                                        :etype etype}
+                                                       em)}))]
+                    (when-let [rule-where (get rule-wheres etype)]
+                      (when (and (not res)
+                                 ;; TODO(dww): remove check once we've figured
+                                 ;;            out how to implement short-circuit?
+                                 (not (:short-circuit? rule-where)))
+                        (tracer/record-info! {:name "instaql/rule-where-failure"
+                                              :attributes {:etype etype
+                                                           :eid eid
+                                                           :rule-where rule-where}})))
+                    res)})])))))
+
+(defn use-rule-wheres? [ctx o]
+  (let [app-id (:app-id ctx)
+        query-hash (forms-hash o)]
+    (flags/use-rule-wheres? {:app-id app-id
+                             :query-hash query-hash})))
+
+(defn rule-wheres->where-conds [etype wheres]
+  (let [forms {etype {:$ {:where wheres}}}]
+    (-> forms
+        ->forms!
+        first
+        :option-map
+        :where-conds)))
+
+(defn get-rule-wheres [ctx rules o]
+  (when (use-rule-wheres? ctx o)
+    (tracer/with-span! {:name "instaql/get-rule-wheres"}
+      (let [{:keys [referenced-etypes]} (instaql-query->patterns ctx o)]
+        (reduce (fn [acc etype]
+                  (if-let [program (rule-model/get-program! rules etype "view")]
+                    (tracer/with-span! {:name "instaql/get-rule-where"
+                                        :attributes {:etype etype
+                                                     :code (:code program)}}
+                      (try
+                        (let [{:keys [short-circuit? where-clauses]}
+                              (cel/get-where-clauses ctx etype (:code program))
+                              ;; convert to where conds here in case this throws
+                              where-conds (rule-wheres->where-conds etype where-clauses)]
+                          (tracer/add-data! {:attributes {:wheres where-clauses
+                                                          :short-circuit? short-circuit?}})
+                          (assoc acc etype {:short-circuit? short-circuit?
+                                            :wheres where-clauses
+                                            :where-conds where-conds}))
+                        (catch Exception e
+                          (tracer/add-data! {:attributes {:exception (.getMessage e)}})
```

Comment:
Curious q: is the reason we don't use `tracer/add-exception!` here, so we don't count this as an exception just yet?